### ISSUE TITLE
babel-plugin-formatjs does not work for FormattedMessage using JSX transform

### ISSUE THREAD

**Issue by felschr** [formatjs/formatjs#3112]
## Which package?
`babel-plugin-formatjs`

## Describe the bug
I have an issue where `babel-plugin-formatjs` doesn't set `id`s for javascript files compiled with `tsc`. Usages of `intl.formatMessage` get updated correctly.
We are using the tsconfig options [`"jsx": "react-jsx"` & `"jsxImportSource": "@emotion/react"`](https://www.typescriptlang.org/tsconfig#jsx), thus a compiled file using `<FormattedMessage>` looks like this:
```js
import { jsx as _jsx, jsxs as _jsxs } from "@emotion/react/jsx-runtime";
import { FormattedMessage } from "react-intl";

export const TestMessage = _jsx(FormattedMessage, { defaultMessage: "Test" }, void 0)
```

It would be great if `babel-plugin-formatjs` could be updated to support usage of `FormattedMessage` via the new (`_jsx`) and old (`React.createElement`) JSX transform.

## To Reproduce
I'll update this once I find the time to create a sample repo.

## Expected behavior
`babel-plugin-formatjs` sets the `id`s for `FormattedMessage` definitions using the new JSX transform (`_jsx`) using various import sources.

## Additional context
I'm working on a monorepo using npm workspaces. The libraries used by the main react client are compiled with `tsc` via their `build` npm script.
We'd like to centralise the automatic id generation in the client project via `babel-plugin-formatjs`, so we don't need to set up `babel-plugin-formatjs` or `@formatjs/ts-transformer` for all library packages of the npm workspaces.

**Comment by longlho** [comment]
if u're using tsc then `@formatjs/ts-transformer`. The transpiled syntax is not stable and is not guaranteed to stay from version to version, also subject to ops like aliasing/hoisting.

**Comment by felschr** [comment]
I've gone the way through `@formatjs/ts-transformer` via `ttypescript` now. I didn't know about `ttypescript` before, it's nice we only need to replace `tsc` with `ttsc` to make it work.
The problem with this setup is, that we can't run the javascript compiled by `ttsc` w/ `@formatjs/ts-transformer` through babel with `babel-plugin-formatjs` anymore. It'll throw a `message.trim is not a function` error now:
```
[1] TypeError: /home/felschr/dev/work/my-project/components/activity/dist/molecules/MyComponent/MyComponent.js: message.trim is not a function
[1] Compiling...
[1] Failed to compile.
```

This likely comes from this line:
https://github.com/formatjs/formatjs/blob/d374ed37e1a126f81259f8a870d844de1869212e/packages/babel-plugin-formatjs/utils.ts#L133

Looking into the code compiled by `ttsc` with `@formatjs/ts-transformer`, `defaultMessage`s now look a little different:
```js
intl.formatMessage({ id: "dFfoZ9", defaultMessage: [{ type: 0, value: "is invalid" }] })
```

Previously without `@formatjs/ts-transformer` it looked like this:
```js
intl.formatMessage({ defaultMessage: "is invalid" })
```

For my use-case we don't need to run babel for those files at the moment, so we simply excluded them again. And I guess if required, it could also be solved by separate babel-loader configurations, so that `babel-plugin-formatjs` doesn't run for code that has already been compiled by `ttsc`. But this definitely seems like a bug that could be solved in `babel-plugin-formatjs`.

While ultimately I think having `babel-plugin-formatjs` support JSX transform directly would be preferable for many setups, the `ttypescript` solution isn't that bad. And if the JSX transform syntax is really too unstable to be supported, I can live with this.
Perhaps the lack of JSX transform support could be highlighted more in the documentation? Issues like these can be quite tricky to debug.

**Comment by longlho** [comment]
if you already use `@formatjs/ts-transformer` why do you need to run it through `babel-plugin-formatjs` again?

**Comment by felschr** [comment]
It would only be a convenience to keep the configuration (e.g. webpack rules) simple.

**Comment by longlho** [comment]
ok we can put in a fix to ignore compiled messages. Do you have a repro setup?

**Comment by gcangela** [comment]
I am also facing the same issue @felschr has. Does anyone have an update on this? It appears that it is trying to trim an array https://github.com/formatjs/formatjs/blob/d374ed37e1a126f81259f8a870d844de1869212e/packages/babel-plugin-formatjs/utils.ts#L133. It works fine for FormattedMessage components, but when using defineMessage or intl.formatMessage it throws the error that `message.trim is not a function`

**Comment by longlho** [comment]
should be fixed in v10.3.8
